tensorly/contrib/decomposition/tt_TTOI.py

```diff
@@ -0,0 +1,188 @@
+
+import numpy as np
+from math import prod
+import tensorly as tl
+from tensorly.tt_tensor import tt_to_tensor, validate_tt_rank
+import warnings
+from tensorly.decomposition._base_decomposition import DecompositionMixin
+
+
+
+def sequential_prod(tensor_prod, multiplier_list, direction):
+    """ Perform sequential multiplication and reshaping
+
+    Parameters
+    ----------
+    tensor_prod: ndrray
+        the tensor to be multiplied by a list of tensors
+        tensor_prod is of dimension r_0*p_1*p_2*...*p_d*r_{d-1}
+    multiplier_list : list 
+        a list of tensors to multiply the tensor
+        len(multiplier_list) <= d-1
+        If direction == "left", multiplier_list[i] is of shape (r_i, p_{i+1}, r_{i+1})
+        If direction == "right", multiplier_list[i] is of shape (r_{d-i}, p_{d-i}, r_{d-i-1})
+    direction : string
+        direction of the multiplication
+    Returns
+    -------
+    tensor_prod : ndarray
+        product tensor
+        If direction == "left" and `multiplier_list` contains N tensors, then `tensor_prod` is of dimension r_N*p_{N+1}*...*p_d*r_{d-1}
+        If direction == "right" and `multiplier_list` contains d-N tensors, then `tensor_prod` is of dimension r_0*p_1*...*p_N*r_N
+    """ 
+
+    if direction == "left":
```

Comment:
Not sure we need a function just for that but if we do decide to keep it, perhaps we can just have left_to_right=True by default and just set it to False to to right to left?

Reply:
Sounds good! I have changed this parameter to left_to_right with the default being True.